### PR TITLE
Add Onboarding Heuristics Tracks Event

### PR DESCRIPTION
**Pull request by adrianduffell**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Adds a new Tracks event to gather heuristics for new onboarding features. I added it to the second step of the OBW because it's on of the first occasions after the user has agreed to tracking. We might want to look at a better way of doing it in the new core profiler

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Pull down this branch on a fresh install of WooCommerce
2. Add a page or post to the WordPress site
3. Add the following Tracks configuration in the dev console: `localStorage.setItem( 'debug', 'wc-admin:*' );`
4. In the Onboarding Wizard, choose any country on the first step
5. On the second step, observe the Tracks event: `wcadmin_onboarding_site_heuristics` contains correct values for `page_count`, `post_count`, and `is_block_theme` values.

### CODE REVIEW

File: plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingSetupWizard.php [hunk 1]
```diff
@@ -230,6 +230,9 @@ public function component_settings( $settings ) {
 		$settings['onboarding']['euCountries']     = WC()->countries->get_european_union_countries();
 		$settings['onboarding']['localeInfo']      = include WC()->plugin_path() . '/i18n/locale-info.php';
 		$settings['onboarding']['profile']         = $profile;
+		$settings['onboarding']['pageCount']       = (int) ( wp_count_posts( 'page' ) )->publish;
```
Comment (moon0326) [inline plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingSetupWizard.php:233]: nitpick: Should we include these values only when users are in the setup wizard by checking `$this->is_setup_wizard()` ?

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀 

I left a very minor comment.

**Review by crunnells — APPROVED** [review]
Tested and LGTM!

File: plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingSetupWizard.php (hunk 1 above)
Comment (adrianduffell) [reply plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingSetupWizard.php:233]: @moon0326  Good idea, I think it's already checked a few lines above? 

https://github.com/woocommerce/woocommerce/blob/483ae18bcac6f32008e03ce9f68666e0562996db/plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingSetupWizard.php#L216-L223

Comment (moon0326) [reply plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingSetupWizard.php:233]: @adrianduffell It does, but it continues if there is a visible task. I thought we could limit the count calls to the OBW. There can be a case where a big store still has tasks left to do. I think this is going to be okay as count calls are cached. I was just being extra cautious as we had a similar issue before.

Comment (adrianduffell) [reply plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingSetupWizard.php:233]: Ah yeah I see what you mean. We could wrap the new properties in `if ( this->is_setup_wizard() ) {}` as a quick follow-up

**Pull request merged**